docs/API-functions.md

```diff
@@ -628,39 +628,110 @@ The following group types are collected in an `SRT_GROUP_TYPE` enum:
+
 } SRT_SOCKGROUPDATA;
 ```
 
 where:
 
 * `id`: member socket ID
-* `status`: current connection status (see `srt_getsockstate`)
-* `result`: result of the operation (if this operation recently updated this structure)
-* `srcaddr`: address to which `id` should be bound
 * `peeraddr`: address to which `id` should be connected
-* `priority`: priority for backup group
+* `sockstate`: current connection status (see `srt_getsockstate`)
+* `memberstate`: current state of the member (see below)
+* `result`: result of the operation (if this operation recently updated this structure)
+
+### SRT_MEMBERSTATUS
+
+The enumeration type that defines the state of the member
+connection in the group:
+
+* `SRT_GST_PENDING`: The connection is in progress, so the socket
+is not currently being used for transmission, even potentially,
+and still has a chance to fail and transit into `SRT_GST_BROKEN`
+without turning into `SRT_GST_IDLE`
+
+* `SRT_GST_IDLE`: The connection is established and ready to
+take over transmission, but it's not used for transmission at
+the moment. This state may last for a short moment in case of
+broadcast or balancing groups. In backup groups this state
+defines a backup link that is ready to take over when the
+currently active (running) link gets unstable.
+
+* `SRT_GST_RUNNING`: The connection is established and at least
+one packet has been alredy sent or received over it.
 
-The priority is set to 0 by default by `srt_prepare_endpoint()` - you can set
-it to a different value afterwards. The default 0 value is the highest priority
-and greater values declare lower priorities. The priority for the backup
-groups determines which link is activated first when the currently active link is
-unstable, and which should keep transmitting when multiple active links are
-currently stable. This is not used by any other group types.
+* `SRT_GST_BROKEN`: The connection got broken. Broken connections
+are not to be revived. Note also that this state is only possible
+to be seen when this state was read by `srt_sendmsg2` or `srt_recvmsg2`,
+just after the failure for this link has been detected, otherwise
+the broken link simply disappears from the member list
 
-Functions to be used on groups:
+Note that this state is only notified as one, however the
+internal member state is separate for sending and receiving.
+There are however some consistency rules: if the state in one
+direction is `RUNNING`, in the other direction it may be `IDLE`,
+and in this case resulting state is `RUNNING`. In all other
+cases the states are the same in both directions.
+
+The states normally should start with `SRT_GST_PENDING` and then
+turn into `SRT_GST_IDLE`. Once the new link is used for sending
+the data, it turns into `SRT_GST_RUNNING`. In case of backup
+groups, if the link was activated, but then it was decided that
+another link needs to remain the only active, the currently
+active link would be "silenced", that is, turn into
+`SRT_GST_IDLE`.
```

Comment:
It actually applies to the member, not to the group (it's speaking about the state array and so it represents the state of every member). What is important here is that this is only one member state representation, while internally the same member has separate states for sending and receiving, so this above intends to describe what rules are used to "merge" these two states into one representation.

Maybe something like this: usually these states are the same in both directions, except a possible situation that the state is `SRT_GST_IDLE` in one direction and `SRT_GST_RUNNING` in the other, in which case the `memberstate` is reported as `SRT_GST_RUNNING`.

Reply:
Better?

Reply:
The latest version is largely refactored:
```
Note that internally the member state is separate for sending and
receiving. If the `memberstate` field of `SRT_SOCKGROUPDATA` is
`SRT_GST_RUNNING`, it means that this is the state in at least one
direction, while in the other direction it may be `SRT_GST_IDLE`. In all
other cases the states should be the same in both directions.

States should normally start with `SRT_GST_PENDING` and then
turn into `SRT_GST_IDLE`. Once a new link is used for sending data, 
the state becomes `SRT_GST_RUNNING`. 
In case of `SRT_GTYPE_BACKUP` type group, if a link is in
`SRT_GST_RUNNING` state, but another link is chosen to remain
as the only active one, this link will be "silenced" (its state will
become `SRT_GST_IDLE`).
```